src/Corax/Queries/TermProviders/TermProvider.NotStartsWith.cs

```diff
@@ -33,10 +35,10 @@ public void Reset()
 
         public bool Next(out TermMatch term)
         {
-
-            while (_iterator.MoveNext(out Slice termSlice, out var _))
+            while (_iterator.MoveNext(out var termScope, out var _))
             {
-                if (termSlice.StartWith(_startWith))
+                var termSlice = termScope.Key.Decoded();
+                if (termSlice.StartsWith(_startWith))
```

Comment:
Can't we do starts with on the encoded value?

Reply:
Pretty sure we can find a way to do it... Yes... haven't focused on that yet.